src/Aws/Common/Credentials/RefreshableInstanceProfileCredentials.php

```diff
@@ -42,6 +42,21 @@ public function __construct(CredentialsInterface $credentials, InstanceMetadataC
         $this->client = $client ?: InstanceMetadataClient::factory();
     }
 
+    public function serialize()
+    {
+        return json_encode(array(
+            'credentials' => parent::serialize(),
+            'client' => serialize($this->client),
```

Comment:
Are you sure that things like the event dispatcher of a client with its anonymous functions can be serialized?

> On Jul 30, 2015, at 5:24 PM, Jonathan Eskew <notifications@github.com> wrote:
> 
> In src/Aws/Common/Credentials/RefreshableInstanceProfileCredentials.php:
> 
> > @@ -42,6 +42,21 @@ public function __construct(CredentialsInterface $credentials, InstanceMetadataC
> >          $this->client = $client ?: InstanceMetadataClient::factory();
> >      }
> >  
> > +    public function serialize()
> > +    {
> > +        return json_encode(array(
> > +            'credentials' => parent::serialize(),
> > +            'client' => serialize($this->client),
> It can. There's no custom serialization defined, so serialize will just capture all object properties. The output is a bit large so I could cut out whatever is unnecessary.
> 
> —
> Reply to this email directly or view it on GitHub.
>